### ISSUE TITLE
Revise cryptograhy dependencies

### ISSUE THREAD

**Issue by postlund** [postlund/pyatv#424]
**What to investigate**
Currently `pyatv` has a bunch of dependencies to 3rd party crypto modules:

* cryptography
  * Pure python: **no**
  * Wheels: **most platforms**
  * Used in:
    * HKDF (SHA512) in `MRP` (`mrp/srp.py`)
    * AES cipher in `AirPlay` (`airplay/srp.py`)
* curve25519-donna
  * Pure python: **no**
  * Wheels: **no**
  * Used in:
    * Key generation in SRP (`airplay/srp.py` and `mrp/srp.py`)
  * Replace with: cryptograhy
    * See: https://cryptography.io/en/latest/hazmat/primitives/asymmetric/x25519/#x25519-key-exchange
* ed25519
  * Pure python: **no**
  * Wheels: **no**
  * Used in:
    * Signing and verification in SPR (`airplay/srp.py` and `mrp/srp.py`)
  * Replace with: cryptography
    * See: https://cryptography.io/en/latest/hazmat/primitives/asymmetric/ed25519/
* tlslite-ng
  * Pure python: **yes**
  * Wheels: **N/A**
  * Used in:
    * Encryption (ChaCha20_Poly1305) in `mrp/chacha20.py`
  * Replace with: cryptograhy
    * See: https://cryptography.io/en/latest/hazmat/primitives/aead/#cryptography.hazmat.primitives.ciphers.aead.ChaCha20Poly1305

I would prefer if these could be consolidated into fewer libraries and also switch to non-pure-python implementations when possible (for performance reasons). Would definately like to switch from tlslite as that impose a lot of start up time, as seen int #420. It is also pure python.

**Expected outcome**
List of libraries that fills the need of `pyatv`.

**Comment by postlund** [comment]
Analysis and prototyping shows that `cryptography` fill all the encryption needs. So hopefully I will be able to drop **three** external dependencies by just using `cryptograhpy` 🎉

**Comment by postlund** [comment]
Some *very* unscientific statistics from doing this change follows below.

**Before:**
```shell
$ time tox -r
real	1m27.506s
user	1m23.776s
sys	0m5.592s
$ time tox
real	0m25.714s
user	0m33.996s
sys	0m1.412s
$ time atvremote -s 10.0.10.81 -n Vardagsrum --mrp-credentials `cat creds` playing
real	0m1.096s
user	0m1.008s
sys	0m0.040s
```

**After:**
```shell
$ time tox -r
real    1m24.099s
user	1m17.984s
sys	0m5.476s
$ time tox
real	0m26.150s
user	0m31.080s
sys	0m1.388s
$ time atvremote -s 10.0.10.81 -n Vardagsrum --mrp-credentials `cat creds` playing
real	0m0.591s
user	0m0.468s
sys	0m0.028s
```
Shaves off a few seconds, not a big deal. But running `atvremote` is almost 50% faster! So it's a good change from a runtime perspective.

**Comment by postlund** [comment]
I have merged this now. A culprit is that a "late enough" OpenSSL must be available, otherwise Ed25519 will fail. I will have to update Troubleshooting with this.
